### PR TITLE
Adds support for per service IAM role session Name

### PR DESCRIPTION
**Pull request by Posnet**
https://github.com/ash2k/iam4kube/issues/4

Having a specific session name for each service account ensures that
it is possible to trace the source API calls in cloudtrails, even if
distinct services share an IAM role.

### CODE REVIEW

**Comment by Posnet** [comment]
I want to add more tests before I merge, but the basics are here.

File: pkg/kube/kroler_test.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	store2 := cache.NewIndexer(mockIndexFunc, cache.Indexers{})
+
+	kroler := Kroler{
+		logger:    logz.DevelopmentLogger(),
+		podIdx:    store,
+		svcAccIdx: store2,
+	}
+
+	role, err := kroler.RoleForIp(context.Background(), "127.0.0.1")
+	assert.Nil(t, role)
```
Comment (ash2k) [inline pkg/kube/kroler_test.go]: By convention if error is not nil other values should not be relied upon. So there is no contract that role will be nil.

File: pkg/kube/kroler_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/ash2k/iam4kube/pkg/util/logz"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/tools/cache"
+)
+
+func mockIndexFunc(obj interface{}) (string, error) {
+	return "foo", nil
+}
+
+func TestRoleForIpNotFound(t *testing.T) {
```
Comment (ash2k) [inline pkg/kube/kroler_test.go]: `t.Parallel()`

File: pkg/kube/BUILD.bazel [hunk 3]
```diff
@@ -14,3 +14,14 @@ go_library(
         "//vendor/k8s.io/client-go/tools/cache:go_default_library",
     ],
 )
+
+go_test(
```
Comment (ash2k) [inline pkg/kube/BUILD.bazel]: Please add size="small" to tell bazel about the test size. And then please run `make fmt-bazel` to add race="on" to enable race detector and format all build files for consistency.

File: pkg/kube/kroler.go [hunk 4]
```diff
@@ -61,6 +63,7 @@ func (k *Kroler) RoleForIp(ctx context.Context, ip iam4kube.IP) (*iam4kube.IamRo
 		return nil, nil
 	}
 	svcAccObj, exists, err := k.svcAccIdx.GetByKey(pod.Namespace + "/" + pod.Spec.ServiceAccountName)
+	fmt.Println(svcAccObj)
```
Comment (ash2k) [inline pkg/kube/kroler.go]: Debugging leftover? :)

File: pkg/kube/kroler_test.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	kroler := newKroler()
+
+	pod := core_v1.Pod{
+		Spec:   core_v1.PodSpec{ServiceAccountName: name},
+		Status: core_v1.PodStatus{PodIP: ipAddr},
+	}
+	pod.Namespace = namespace
+	kroler.podIdx.Add(&pod)
+
+	svcAccount := core_v1.ServiceAccount{}
```
Comment (ash2k) [inline pkg/kube/kroler_test.go]: Any particular reason to initialize it this way? In Go the convention is to use literals and embed field initialization like that:
```go
svcAccount := core_v1.ServiceAccount{
    Name: name,
    Namespace: namespace,
    Annotations: map[string]string{
        iam4kube.IamRoleArnAnnotation: testArn,
    },
}
```
and so on...

File: pkg/kube/kroler_test.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+	svcAccount.Annotations = make(map[string]string)
+	svcAccount.Annotations[iam4kube.IamRoleArnAnnotation] = testArn
+	kroler.svcAccIdx.Add(&svcAccount)
+
+	expectedRole := &iam4kube.IamRole{}
+	expectedRole.Arn = arn.ARN{"aws", "iam", "", "123456789012", "role/test_role"}
+	expectedRole.SessionName = namespace + "/" + name
+
+	role, err := kroler.RoleForIp(context.Background(), iam4kube.IP(ipAddr))
+	assert.Nil(t, err)
```
Comment (ash2k) [inline pkg/kube/kroler_test.go]: `require.NoError(t, err)`. `require.*` methods abort test execution on failure by unwinding the stack, executing all `defer`ed dunctions. I.e. you don't want to proceed if `err` is not `nil`.  And `NoError` will give a more meaningful message on failure.

File: pkg/kube/kroler_test.go [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+	namespace := "Foo"
+	name := "Bar"
+	ipAddr := "127.0.0.1"
+
+	kroler := newKroler()
+
+	pod := core_v1.Pod{
+		Spec:   core_v1.PodSpec{ServiceAccountName: name},
+		Status: core_v1.PodStatus{PodIP: ipAddr},
```
Comment (ash2k) [inline pkg/kube/kroler_test.go]: Please wrap lines here, no need to write it in one line.

**Pull request merged**